Taskfile.yml

```diff
@@ -261,6 +272,30 @@ tasks:
     env:
       RECORD_REPLAY: 0
 
+  controller:test-integration-kind:
+    desc: Run live integration tests in kind.
+    dir: "{{.CONTROLLER_ROOT}}"
+    deps: [controller:kind-create-with-podidentity]
+    cmds:
+      # This timeout is purposefully low - if we find that this job is taking a long time we may need to think of other ways
+      # to keep CI fast
+      # TODO: Below pattern was taken from https://github.com/go-task/task/pull/474 to ensure cleanup runs after the test.
+      - |
+        (go test -timeout 15m -run '{{default ".*" .TEST_FILTER}}' ./test); # Run the tests in a subshell
+        TEST_EXIT=$?;
+        kind delete cluster --name=asov2
```

Comment:
Maybe split the cluster deletion out so that it's possible to run `test-integration-kind` without tearing down an existing kind cluster? Although I guess this way prevents having old CRDs accidentally hanging around.

Reply:
The problem is that if the tests fail, there is no way to write a task in such a way that other tasks "run afterwards". So you have to do the tests + delete in the same task in order to deal with the fail case. In our other tests that's not an issue because there's not a long-running thing (the kind cluster) we want to avoid leaking.

I think at least for the purposes of this PR, keeping it as is makes sense for this target.

I added a separate target `controller:test-integration-kind` and renamed this target to `controller:test-integration-kind-ci`. Also added a note on both to explain what they're for a bit more clearly. I think that accomplishes what you want while maintaining the cleanup step in CI where we need it.

Reply:
Yup, that makes sense, thanks!